### ISSUE TITLE
Test adaptive training loop

### ISSUE THREAD

**Issue by tgvp** [tgvp/Machine-Learning#3]
Test new function added to adjust learning rate.

Consider implementing early stopping in **Dev** branch.

Depending on results, retrain the model and evaluate.
